Index: common.php
===================================================================
--- common.php	(revision 12170)
+++ common.php	(working copy)
@@ -133,7 +133,14 @@
         $sourcefrom="install";
     }
 } else {
-    $sourcefrom="public";
+	if (strpos($slashlesspath, "user"))
+	{
+		$sourcefrom="admin";
+	}
+	else
+	{
+		$sourcefrom="public";
+	}
 }
 
 // Set path for captcha verification.php
Index: admin/admin.php
===================================================================
--- admin/admin.php	(revision 12170)
+++ admin/admin.php	(working copy)
@@ -76,6 +76,11 @@
 
 if(isset($_SESSION['loginID']))
 {
+	if (!$_SESSION['USER_RIGHT_SUPERADMIN'])
+	{
+		print ('<script type="text/javascript">window.open(\'../user\', \'_top\');</script>');
+	}
+	
     //VARIOUS DATABASE OPTIONS/ACTIONS PERFORMED HERE
     if (in_array($action, array('updateemailtemplates','delsurvey','delgroup','delquestion','insertsurvey','updatesubquestions','copynewquestion','insertquestiongroup','insertCSV','insertquestion','updatesurveysettings','updatesurveysettingsandeditlocalesettings','updatesurveylocalesettings','updategroup','deactivate','savepersonalsettings','updatequestion','updateansweroptions','renumberquestions','updatedefaultvalues')))
     {
